### ISSUE TITLE
[EXPORT] Video delay

### ISSUE THREAD

**Issue by Lopolin-LP** [olive-editor/olive#1625]
**Commit Hash** 
72d5b4af

**Platform** 
Windows 10

**Summary**
There's a video delay, if the scene changes rapidly _(example: If you enter a different dimension in Minecraft, and the loading screen is only a few frames long, Olive _may_ struggle to keep up)_. Because nobody _seems_ (I have searched for keywords in open and closed issues) to have reported this issue, im probably the only one who has noticed this issue, when a rapid scene change happens. This doesn't happen whilst in the editor, but is there, after the Project got exported.

**Additional Information / Output**
- Screensaver doen't influence this.
- Happens with the file format MPEG-4 Video (mp4) and QuickTime (mov).
- The Quality of the Video doesn't influence this.
- If everything is set to the preset, this still happens.
- The Original 21 Minute mkv-File has no Offset.
- I haven't tried 0.1 _yet._
- Olive has been forced to use the GPU. (Options: System > Display > Graphics Options) (Because I have installed a new Version, and haven't readded it there, it probably wasn't forced)
- Priority (via Taskmanager) has probably no influence.
- I have Layered a Video (it's also an mkv-File) over the other one... now theres an empty "Video 1" (Video 1 to 3 and Audio 1 to 2). The Video that has been Layered over the other one is tiny, whilst having a high resolution of 1920 x 1080.
- **About the rapid scene Change:** The Video "Lag" (Offset) happens after I was teleported to the... "Battle Field". There was a Loading screen for about 12 Frames during the Teleport _(9 had a purple-red background and 3 had a brown background)_. The Offset even increses the longer I am in the Video. Probably after I turn very fast or (maybe) open my inventory or any other Inventory (and that happens a **LOT**).

PC specs:
- CPU: AMD Ryzen 5 3600 6-Core Processor
- RAM: 16 GB
- GPU: NVIDIA GeForce GTX 1050 Ti

Am I the guy who's stupid (Im fairly new to video editing) or is Olive the thing that causes my issue? Or is it something entirely different?
Ask Questions if I am too confusing.

**Comment by itsmattkc** [comment]
Can you provide some examples and perhaps some screenshots/screen recordings? "Video delay" isn't very descriptive.

**Comment by itsmattkc** [comment]
Interesting, would you be able to provide the source video file?

**Comment by TheTechRobo** [comment]
I thought they gave it:

> Cropped (with Windows' Video-Editor) version (720p), no other edits (everything is fine): mediafire.com/file/v4za83630gia545/2021-05-10_Hive_Speedrun_4_36.185.mp4/file

**Comment by itsmattkc** [comment]
Is that the source file that was imported into Olive? I was under the impression that was just another export for comparison demonstrating that other software doesn't have this issue.

**Comment by TheTechRobo** [comment]
I won't speak for them, but I believe so.

> Cropped (**with Windows' Video-Editor**) version (720p), no other edits (everything is fine): mediafire.com/file/v4za83630gia545/2021-05-10_Hive_Speedrun_4_36.185.mp4/file

**Comment by Lopolin-LP** [comment]
> Is that the source file that was imported into Olive? I was under the impression that was just another export for comparison demonstrating that other software doesn't have this issue.

Yes, the other file is a other export from windows' build in video-editor. I didn't feel like releasing the full 21 minute uncensored video, bcs there _might_ be something I don't want to show and I don't wanna risk it...

also: the file that is from Windows' built-in video-editor is just cropped, no other edits. (Over at Olive I have re-timed the "Skywars speedrun" because I have screwed it up.)

**Comment by itsmattkc** [comment]
> also: the file that is from Windows' built-in video-editor is just cropped, no other edits. (Over at Olive I have re-timed the "Skywars speedrun" because I have screwed it up.)

I can test it, but there's a good chance that since the footage has been re-encoded through other software, it won't exhibit the same issue. I'll let you know.

**Comment by benlorenz** [comment]
I just observed something similar when exporting a video (basically just cutting out about 30 seconds from a video from a dslr + some transitions at the beginning and the end) with some spoken voice, somewhere in the middle the video/audio goes out of sync (slowly) and the lips do not match the audio anymore (no cuts in between). (Unfortunately the video is rather personal so I cannot post it here)
It did happen both with AAC audio and MP3 audio and even when exporting video and audio separately and merging them with ffmpeg afterwards.

Since both this issue and my olive build were very recent I took wild guess and reverted 15556c64aea5cf9494c97b43f4a3ad7aed4ea83d which seems to have fixed that issue for me. I did several exports again and got no audio delay anymore.
